src/components/node-text.js

```diff
@@ -15,51 +15,83 @@
   limitations under the License.
 */
 
-import * as React from 'react';
+import React, { useMemo, useEffect, useRef, useCallback } from 'react';
+import * as d3 from 'd3';
 import GraphUtils from '../utilities/graph-util';
 import { type INode } from './node';
+import {
+  DEFAULT_NODE_TEXT_MAX_TITLE_CHARS,
+  DEFAULT_NODE_TEXT_LINE_OFFSET,
+} from '../constants';
 
 type INodeTextProps = {
   data: INode,
   nodeTypes: any, // TODO: create a nodeTypes interface
   isSelected: boolean,
-  maxTitleChars: number,
+  maxTitleChars?: number,
+  lineOffset?: number,
 };
 
-class NodeText extends React.Component<INodeTextProps> {
-  getTypeText(data: INode, nodeTypes: any) {
-    if (data.type && nodeTypes[data.type]) {
-      return nodeTypes[data.type].typeText;
-    } else if (nodeTypes.emptyNode) {
-      return nodeTypes.emptyNode.typeText;
-    } else {
-      return null;
-    }
+function getTypeText(data: INode, nodeTypes: any) {
+  if (data.type && nodeTypes[data.type]) {
+    return nodeTypes[data.type].typeText;
+  } else if (nodeTypes.emptyNode) {
+    return nodeTypes.emptyNode.typeText;
+  } else {
+    return null;
   }
+}
 
-  render() {
-    const { data, nodeTypes, isSelected, maxTitleChars } = this.props;
-    const lineOffset = 18;
-    const title = data.title;
-    const className = GraphUtils.classNames('node-text', {
-      selected: isSelected,
-    });
-    const typeText = this.getTypeText(data, nodeTypes);
+function NodeText({
+  data,
+  nodeTypes,
+  isSelected,
+  maxTitleChars = DEFAULT_NODE_TEXT_MAX_TITLE_CHARS,
+  lineOffset = DEFAULT_NODE_TEXT_LINE_OFFSET,
+}: INodeTextProps) {
+  const nodeTextRef = useRef();
 
-    return (
-      <text className={className} textAnchor="middle">
-        {!!typeText && <tspan opacity="0.5">{typeText}</tspan>}
-        {title && (
-          <tspan x={0} dy={lineOffset} fontSize="10px">
-            {title.length > maxTitleChars
-              ? title.substr(0, maxTitleChars)
-              : title}
-          </tspan>
-        )}
-        {title && <title>{title}</title>}
-      </text>
-    );
-  }
+  const title = data.title;
+  const className = useMemo(
+    () =>
+      GraphUtils.classNames('node-text', {
+        selected: isSelected,
+      }),
+    [isSelected]
+  );
+
+  // prevents the SVG click event from firing when the node text is selected
+  const handleTextClick = useCallback((event: any) => {
+    event.stopPropagation();
+  }, []);
+
+  const typeText = useMemo(() => getTypeText(data, nodeTypes), [
+    data,
+    nodeTypes,
+  ]);
+
+  useEffect(() => {
+    d3.select(nodeTextRef.current).on('click', () => handleTextClick(d3.event));
+  }, [handleTextClick]);
+
+  return (
+    // Note: when "is" is used, we must use real HTML attributes, like text-anchor, not React-based attributes
+    <text ref={nodeTextRef} class={className} text-anchor="middle" is="text">
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/is
why are we using "is"? it seems to be meant for Web Component (which we're not using here) and breaks some stuff based on the comments

Reply:
Enzyme doesn't recognize SVG elements so you have to use "is" to trick it into rendering the element as a normal HTML element. You can write anything in the "is" value but the common use is to rewrite the element name itself.

```
Warning: The tag <use> is unrecognized in this browser. If you meant to render a React component, start its name with an uppercase letter.
```

https://stackoverflow.com/questions/48751919/how-can-i-suppress-the-tag-some-tag-is-unrecognized-in-this-browser-warning/55537927#55537927